lib/bolt/config.rb

```diff
@@ -34,50 +39,75 @@ class Config
+    # Options that configure Bolt. These options are used in bolt.yaml and
+    # bolt-defaults.yaml.
+    BOLT_CONFIG = {
+      "color"               => "Whether to use colored output when printing messages to the console.",
+      "compile-concurrency" => "The maximum number of simultaneous manifest block compiles.",
+      "concurrency"         => "The number of threads to use when executing on remote targets.",
+      "format"              => "The format to use when printing results. Options are `human` and `json`.",
+      "plugin_hooks"        => "Which plugins a specific hook should use.",
+      "plugins"             => "A map of plugins and their configuration data.",
+      "puppetdb"            => "A map containing options for configuring the Bolt PuppetDB client.",
+      "puppetfile"          => "A map containing options for the `bolt puppetfile install` command.",
+      "save-rerun"          => "Whether to update `.rerun.json` in the Bolt project directory. If "\
+                               "your target names include passwords, set this value to `false` to avoid "\
+                               "writing passwords to disk."
+    }.freeze
 
-    DEFAULT_TRANSPORT_OPTION = { 'transport' => 'ssh' }.freeze
+    # These options are only available to bolt-defaults.yaml.
+    DEFAULTS_CONFIG = {
+      "inventory-config" => "A map of default configuration options for the inventory. This includes options "\
+                            "for setting the default transport to use when connecting to targets, as well as "\
+                            "options for configuring the default behavior of each transport."
+    }.freeze
 
-    CONFIG_IN_INVENTORY = TRANSPORT_CONFIG.merge(TRANSPORT_OPTION)
+    # Options that configure the inventory, specifically the default transport
+    # used by targets and the transports themselves. These options are used in
+    # bolt.yaml, inventory.yaml, and under the inventory-config key in
+    # bolt-defaults.yaml.
+    INVENTORY_CONFIG = {
+      "docker"    => "A map of configuration options for the docker transport.",
+      "local"     => "A map of configuration options for the local transport.",
+      "pcp"       => "A map of configuration options for the pcp transport.",
+      "remote"    => "A map of configuration options for the remote transport.",
+      "ssh"       => "A map of configuration options for the ssh transport.",
+      "transport" => "The default transport to use when the transport for a target is not specified in the URI.",
+      "winrm"     => "A map of configuration options for the winrm transport."
+    }.freeze
 
-    # NOTE: All configuration options should have a corresponding schema property
-    #       in schemas/bolt-config.schema.json
-    OPTIONS = {
+    # Options that configure the project, such as paths to files used for a
+    # specific project. These settings are used in bolt.yaml and bolt-defaults.yaml.
+    PROJECT_CONFIG = {
```

Comment:
Now that `bolt-project.yaml` is one file, I think we can move the `Bolt::Project::PROJECT_SETTINGS` values here as well and remove that constant from Bolt::Project

Reply:
I think this change should be made as part of the config references doc changes, since it's currently used by that page to create a separate section for those options.

Reply:
Hm, that makes me wonder if we should make the bolt.yaml configuration static so that we can update these without having to worry about legacy docs?